src/main/java/seedu/address/model/FriendlyLink.java

```diff
@@ -387,6 +363,103 @@ public ObservableList<Pair> getPairList() {
         return pairs.asUnmodifiableObservableList();
     }
 
+    /**
+     * Checks whether regions of an elderly and a volunteer are the same.
+     * If null is specified for any Nric, checks are conducted on all involved pairs.
+     * The elderly and volunteer with the given Nric must exist in FriendlyLink.
+     *
+     * @param elderlyNric Nric of the elderly.
+     * @param volunteerNric Nric of the volunteer.
+     * @return True if both persons of all involved pairs belong in the same region, false otherwise.
+     */
+    public boolean checkIsSameRegion(Nric elderlyNric, Nric volunteerNric) {
+        return checkPairs(elderlyNric,
+                volunteerNric, Person::isSameRegion);
+    }
+
+    /**
+     * Checks whether there are suitable available dates between an elderly and a volunteer.
+     * If null is specified for any Nric, checks are conducted on all involved pairs.
+     * The elderly and volunteer with the given Nric must exist in FriendlyLink.
+     *
+     * @param elderlyNric Nric of the elderly.
+     * @param volunteerNric Nric of the volunteer.
+     * @return True if both persons share common available dates or at least one person
+     *     has no specified available dates for all involved pairs, false otherwise.
+     */
+    public boolean checkHasSuitableAvailableDates(Nric elderlyNric, Nric volunteerNric) {
+        return checkPairs(elderlyNric,
+                volunteerNric, Person::hasSuitableAvailableDates);
+    }
+
+    /**
+     * Checks whether pairs with specified elderly and volunteer satisfies a given predicate.
+     * If null is specified for any Nric, checks are conducted on all involved pairs.
+     * The elderly and volunteer with the given Nric must exist in FriendlyLink.
+     *
+     * @param elderlyNric Nric of the elderly.
+     * @param volunteerNric Nric of the volunteer.
+     * @param predicate Predicate to check pairs.
+     * @return True if all involved pairs satisfies the given predicate, false otherwise.
+     */
+    private boolean checkPairs(Nric elderlyNric, Nric volunteerNric,
+            BiFunction<Elderly, Volunteer, Boolean> predicate) {
+        List<Elderly> elderlyToCheck = elderlyNric == null
+                ? getPairedElderly(volunteerNric)
+                : new ArrayList<>(Collections.singletonList(getElderly(elderlyNric)));
+        List<Volunteer> volunteerToCheck = volunteerNric == null
+                ? getPairedVolunteers(elderlyNric)
+                : new ArrayList<>(Collections.singletonList(getVolunteer(volunteerNric)));
+        for (Elderly elderly : elderlyToCheck) {
+            for (Volunteer volunteer : volunteerToCheck) {
+                if (!predicate.apply(elderly, volunteer)) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
```

Comment:
Will there be a use case where both nric is `null`? If not, can you add an assertion here to ensure that? Also, I feel like the different use cases can be abstracted into two functions instead, but that's up to you.

Reply:
Both nric is null can be used to check for all pairs in friendlylink, which might be useful in the future if we add other commands that can potentially edit multiple entities at once

I have considered abstracting this out to different functions, but it appears to have too much repeated functionality since  `checkIsSameRegion` and `checkHasSuitableAvailableDates` would both require null and non null versions for elderly and volunteer

Reply:
Right now, `getPairedVolunteers` and `getPairedElderly` returns an empty list when `nric` is null. Is this intended?